Task/strip stake address prefix

Dodilanne opened the pull request:
# Problems
1. The anvil backend always returns stake keys without the `e1` or `e0` prefix but the wallet extensions return them with the prefix. Solution: I added a stakeAddressHexPrefixed to the wallet store.
2. The previous implementation of the `hexToBech32` function used the starting characters of the hex value to determine if a prefix had to be appended which is wrong because some stake addresses actually start with `e0` or `e1`. For example, this is a valid and existing stake address in hex format without any prefix: `e09fd385ca1eb59f37c18f1e6b70d21cb4c4d7fd31eafd42ac90118f`. Solution: Use the length of the hex value to determine if a prefix is present or not instead of the starting characters.

jbergeron03 reviewed (changes requested):
The same logic applies to the `changeAddress`, so we should aim for consistency in our approach.

I suggest we either:

- Standardize the format by always prefixing it, and provide a utility function for users to strip the prefix if needed.
- Alternatively, also expose the `changeAddress` with the prefix in hex format.

I personally lean toward the first option but am open to discussion.

Here's a link to compare the changeAddress on [preprod](https://preprod.cardanoscan.io/address/00495d6856a5c2e172c4e046bae623db6278180a9ad58b9690d046cc5969390791217ad3ed6790b3f369f90d135b8f1a3e8a1947b4f8ed146d) and [prod](https://cardanoscan.io/address/01495d6856a5c2e172c4e046bae623db6278180a9ad58b9690d046cc5969390791217ad3ed6790b3f369f90d135b8f1a3e8a1947b4f8ed146d)

Dodilanne commented:
I implemented the first approach you suggested: the store always returns prefixed addresses and I exposed a `stripAddressPrefix` util to facilitate conversion.

As we discussed, I also removed the code that made sure a prefix was always set on hex stake addresses since it was only a patch to handle inconsistent return values returned by internal anvil backend logic.

jbergeron03 reviewed (approved):
👌👌👌
